faucet/src/main.rs

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use axum::{
+    error_handling::HandleErrorLayer,
+    http::StatusCode,
+    response::IntoResponse,
+    routing::{get, post},
+    BoxError, Extension, Json, Router,
+};
+use std::{borrow::Cow, net::SocketAddr, sync::Arc, time::Duration};
+use sui::{
+    sui_config_dir,
+    wallet_commands::{WalletCommands, WalletContext},
+    SUI_WALLET_CONFIG,
+};
+use sui_faucet::{Faucet, FaucetRequest, FaucetResponse, SimpleFaucet};
+use tower::ServiceBuilder;
+use tracing::info;
+
+const DEFAULT_AMOUNT: u64 = 20;
+const DEFAULT_NUM_COINS: usize = 5;
+const REQUEST_BUFFER_SIZE: usize = 10;
+// TODO: Increase this once we use multiple gas objects
+const CONCURRENCY_LIMIT: usize = 1;
+const TIMEOUT_IN_SECONDS: u64 = 120;
+
+struct AppState<F = SimpleFaucet> {
+    faucet: F,
+    // TODO: add counter
+}
```

Comment:
In case I want to experiment with a different implementation of Faucet(e.g., using a queue of gas objects,etc) for some traffic and can quickly switch to another one in case the more complicated implementation is failing.